.github/workflows/dirty-waters.yml

```diff
@@ -0,0 +1,49 @@
+name: Dirty Waters
+on:
+  pull_request: ~
+  push:
+    branches:
+      - main
+  schedule:
+    - cron: "0 3 * * *"
+  workflow_dispatch: ~
+
+permissions: read-all
+
+jobs:
+  analyze:
+    name: Analysis
+    runs-on: ubuntu-24.04
+    permissions:
+      pull-requests: write # To comment on a Pull Request
+    steps:
+      - name: Harden runner
+        uses: step-security/harden-runner@0080882f6c36860b6ba35c610c98ce87d4e2f26f # v2.10.2
+        with:
+          egress-policy: block
+          allowed-endpoints: >
+            api.github.com:443
+            azure.archive.ubuntu.com:80
+            esm.ubuntu.com:443
+            files.pythonhosted.org:443
+            github.com:443
+            packages.microsoft.com:443
+            pypi.org:443
+            raw.githubusercontent.com:443
+            registry.npmjs.org:443
+      - name: Checkout repository
+        uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938 # v4.2.0
```

Comment:
This version **MUST** be synced with the rest of the project before this is merged, see https://github.com/search?q=repo%3Aericcornelissen%2Fshescape+actions%2Fcheckout&type=code

Reply:
Resolved with 734805298dc2a376df6ea86795c9e839d44a148e